pydata_google_auth/cli.py

```diff
@@ -0,0 +1,55 @@
+"""Private module that implements a pydata-google-auth CLI tool."""
+
+import argparse
+import sys
+
+
+LOGIN_HELP = (
+    "Login to Google and save user credentials as a JSON file to use as "
+    "Application Default Credentials."
+)
+LOGIN_SCOPES_DEFAULT = "https://www.googleapis.com/auth/cloud-platform"
+LOGIN_SCOPES_HELP = (
+    "Comma-separated list of scopes (permissions) to request from Google. "
+    "See: https://developers.google.com/identity/protocols/googlescopes for "
+    "a list of available scopes. Default: {}"
+).format(LOGIN_SCOPES_DEFAULT)
+LOGIN_CLIENT_ID_HELP_TEMPLATE = (
+    "(Optional, but recommended) Client {}. Use this in combination with "
+    "the {other} argument to authenticate with an application other than the "
+    "default (PyData Auth). This argument is required to use APIs the track "
+    "billing and quotas via the application (such as Cloud Vision), rather "
+    "than billing the user (such as BigQuery does)."
+)
+LOGIN_CLIENT_ID_HELP = LOGIN_CLIENT_ID_HELP_TEMPLATE.format(
+    "ID", other="--client-secret"
+)
+LOGIN_CLIENT_SECRET_HELP = LOGIN_CLIENT_ID_HELP_TEMPLATE.format(
+    "secret", other="--client-id"
+)
+
+
+def main():
+    parser = argparse.ArgumentParser(description=__doc__)
+    subparsers = parser.add_subparsers(title="commands", dest="command")
+
+    login_parser = subparsers.add_parser("login", help=LOGIN_HELP)
+    login_parser.add_argument(
+        "destination", help="Path to save user credentials JSON file to."
+    )
+    login_parser.add_argument(
+        "--scopes", help=LOGIN_SCOPES_HELP, default=LOGIN_SCOPES_DEFAULT
+    )
+    login_parser.add_argument("--client-id", help=LOGIN_CLIENT_ID_HELP)
+    login_parser.add_argument("--client-secret", help=LOGIN_CLIENT_SECRET_HELP)
+
+    args = parser.parse_args()
+    if args.command == "login":
+        pass  # TODO: continue parsing scopes and then call save user creds.
+    else:
+        print('Got unknown command "{}".'.format(args.command), file=sys.stderr)
+        parser.print_help()
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
Can also call this file `__main__.py` and then `python -m pydata-google-auth` calls this directly
(medium confidence - maybe I'm missing something because I still see this pattern in lots of well-designed libraries - please let me know if I am)

Reply:
Good call. I think it's fine to just have `python -m pydata_google_auth` without the `console_scripts`.